### PR TITLE
 Remove broken and deprecated reference to DebuggerManager

### PR DESCRIPTION
**Pull request by metatablecat**
## Changes

Removes a broken reference to the deprecated and mostly obsolete DebuggerManager.

## Checks

By submitting your pull request for review, you agree to the following:

- [x] This contribution was created in whole or in part by me, and I have the right to submit it under the terms of this repository's open source licenses.
- [x] I understand and agree that this contribution and a record of it are public, maintained indefinitely, and may be redistributed under the terms of this repository's open source licenses.
- [x] To the best of my knowledge, all proposed changes are accurate.

### CODE REVIEW

**Comment by metatablecat** [comment]
The best approach really is to merge DebuggerManager.yaml from the old documentation so that the link fixes, but its so obsolete to the point where even Roblox's Luau LSP doesn't know about it, its probably not worth it.

Converting to draft so that I can make a mega commit removing all references, or changing them to just `DebuggerManager`.

**Comment by metatablecat** [comment]
Formatting may need to be fixed on this commit as I did it on a phone.

File: content/en-us/reference/engine/globals/RobloxGlobals.yaml [hunk 1]
```diff
@@ -114,17 +114,23 @@ functions:
...
+      Refers to the legacy DebuggerManager, which acts as an interface for
+      the Luau debugger feature.
     description: |
-      Returns the `Class.DebuggerManager` class, which acts as an interface for
-      Roblox's Lua debugger feature.
+      Returns the legacy `DebuggerManager` class, which acts as an interface
+      for Roblox's Luau debugger feature.
+
+      This function is not recognised by Luau's analysis tool and will raise
+      an undefined global warning.
```
Comment (IgnisRBX) [inline content/en-us/reference/engine/globals/RobloxGlobals.yaml:124]:
```suggestion
      Returns the legacy `DebuggerManager` class which acts as an interface
      for Roblox's Luau debugger feature.

      This function is not recognized by Luau's analysis tool and will raise
      an undefined global warning.
```

File: content/en-us/reference/engine/globals/RobloxGlobals.yaml [hunk 2]
```diff
@@ -114,17 +114,23 @@ functions:
...
     parameters:
     returns:
       - type: DebuggerManager
         summary: ''
     tags:
       - Deprecated
+    deprecation_message: |
+      The `DebuggerManager` is obsolete and serves little to no usecase to
+      developers.
```
Comment (IgnisRBX) [inline content/en-us/reference/engine/globals/RobloxGlobals.yaml:133]:
```suggestion
      The `DebuggerManager` is obsolete and serves little to no use case for
      developers.
```

**Review by IgnisRBX — APPROVED** [review]

**Pull request merged**